### DISCUSSION TITLE
Do we need to handle 1000+ identity providers on login page?

### DISCUSSION THREAD

**Discussion by mposolda** [keycloak/keycloak#31975] [Feedback]
### Description

There are some use-cases (I am aware at least 1 use-case from our community) where people need to handle many identity providers on the login page.

There is some switch like `Hide on login page` on identity provider configuration, but sometimes people may still want the Idps to be available.

For instance: Do we want users to be able to "filter" Idps by themselves on the login page?

**Comment by cgeorgilakis** [comment]
Filtering is needed for 1000+ IdPs. However, the most severe problem is that login page will become unresponsive for so many IdPs without paging.

I do not know if and how ftl can handle pager with search. In our theme, we have developed a complex solution with angularjs with pager and search. 

We believe that [Keycloak: Home IdP Discovery](https://github.com/sventorben/keycloak-home-idp-discovery) might be a solution. Unfortunately, we do not have time now to investigate further it.

**Comment by sguilhen** [comment]
@mposolda Can we turn this into a discussion, and then open issues once we have a clear path of what we want to do with the login page when there are thousands of IDPs in a realm?

I personally think it is not good to present users with a huge number o IDPs to chose from, looks like bad UX to me. With organizations we expect to see only a few IDPs available for login once we identify the orgs a user belongs to. 

I don't know exactly how people are doing it with their use cases without orgs - @cgeorgilakis mentioned he had to develop a custom solution to allow for paging/searching IDPs during login in, but I would expect others to use things idp_hint or perhaps the Home IdP Discovery linked above (I haven't checked it in detail yet) to narrow the IDPs down to a reasonable amount.

If the result of the discussion points to the need of really supporting thousands of IDPs during login with pagination/search, then I would open issues specifically for that.

**Comment by pedroigor** [comment]
I agree, we discussed this with @stianst and I think we can close this issue and open a discussion because the scope of the parent epic does not involve any change to the login page to support such a amount of brokers. The reason is that we now have organizations that provide a solution for those using today multiple brokers when addressing B2B or multi-tenancy related use cases.

**Comment by darius-m** [comment]
In research and education there are large public user federations (e.g., [Edugain](https://edugain.org/)) that, if supported, require allowing users to pick from a very large number of identity provides (as well as registering / updating SAML clients).
